src/uu/factor/src/factor.rs

```diff
@@ -3,302 +3,147 @@
 // For the full copyright and license information, please view the LICENSE
 // file that was distributed with this source code.
 
-#![allow(clippy::items_after_test_module)]
-use smallvec::SmallVec;
-use std::cell::RefCell;
-use std::fmt;
-
-use crate::numeric::{Arithmetic, Montgomery};
-use crate::{miller_rabin, rho, table};
-
-type Exponent = u8;
-
-#[derive(Clone, Debug, Default)]
-struct Decomposition(SmallVec<[(u64, Exponent); NUM_FACTORS_INLINE]>);
-
-// spell-checker:ignore (names) Erdős–Kac * Erdős Kac
-// The number of factors to inline directly into a `Decomposition` object.
-// As a consequence of the Erdős–Kac theorem, the average number of prime factors
-// of integers < 10²⁵ ≃ 2⁸³ is 4, so we can use a slightly higher value.
-const NUM_FACTORS_INLINE: usize = 5;
-
-impl Decomposition {
-    fn one() -> Self {
-        Self::default()
-    }
-
-    fn add(&mut self, factor: u64, exp: Exponent) {
-        debug_assert!(exp > 0);
-
-        if let Some((_, e)) = self.0.iter_mut().find(|(f, _)| *f == factor) {
-            *e += exp;
-        } else {
-            self.0.push((factor, exp));
-        }
-    }
-
-    #[cfg(test)]
-    fn product(&self) -> u64 {
-        self.0
-            .iter()
-            .fold(1, |acc, (p, exp)| acc * p.pow(*exp as u32))
-    }
-
-    fn get(&self, p: u64) -> Option<&(u64, u8)> {
-        self.0.iter().find(|(q, _)| *q == p)
-    }
+// spell-checker:ignore funcs
+
+use std::collections::BTreeMap;
+use std::io::BufRead;
+use std::io::{self, stdin, stdout, Write};
+
+use clap::{crate_version, Arg, ArgAction, Command};
+use num_bigint::BigUint;
+use num_traits::FromPrimitive;
+use uucore::display::Quotable;
+use uucore::error::{set_exit_code, FromIo, UResult, USimpleError};
+use uucore::{format_usage, help_about, help_usage, show_error, show_warning};
+
+const ABOUT: &str = help_about!("factor.md");
+const USAGE: &str = help_usage!("factor.md");
+
+mod options {
+    pub static EXPONENTS: &str = "exponents";
+    pub static HELP: &str = "help";
+    pub static NUMBER: &str = "NUMBER";
 }
 
-impl PartialEq for Decomposition {
-    fn eq(&self, other: &Self) -> bool {
-        for p in &self.0 {
-            if other.get(p.0) != Some(p) {
-                return false;
-            }
-        }
-
-        for p in &other.0 {
-            if self.get(p.0) != Some(p) {
-                return false;
-            }
+fn print_factors_str(
+    num_str: &str,
+    w: &mut io::BufWriter<impl io::Write>,
+    print_exponents: bool,
+) -> UResult<()> {
+    let x = match num_str.trim().parse::<num_bigint::BigUint>() {
+        Ok(x) => x,
+        Err(e) => {
+            // We return Ok() instead of Err(), because it's non-fatal and we should try the next
+            // number.
+            show_warning!("{}: {}", num_str.maybe_quote(), e);
+            set_exit_code(1);
+            return Ok(());
         }
+    };
```

Comment:
```suggestion
let x = num_str.trim().parse::<num_bigint::BigUint>().map_err(|e| {
    show_warning!("{}: {}", num_str.maybe_quote(), e);
    set_exit_code(1);
    Ok(())
})?;
```

Reply:
the indentation is incorrect :)

Reply:
sorry, I can't make this compile on my machine.

```
error[E0277]: the trait bound `Result<(), _>: UError` is not satisfied
  --> src/uu/factor/src/factor.rs:37:7
   |
37 |     })?;
   |       ^ the trait `UError` is not implemented for `Result<(), _>`
   |
   = help: the following other types implement trait `UError`:
             BackupError
             FormatError
             USimpleError
             UUsageError
             UIoError
             uucore::error::ExitCode
             ClapErrorWrapper
   = note: required for `Box<dyn UError>` to implement `From<Result<(), _>>`
   = note: required for `Result<(), Box<dyn UError>>` to implement `FromResidual<Result<Infallible, Result<(), _>>>`
```

I tried different variants, but I still think the original implementation is best.

Reply:
check deleted file `cli.rs` there you can find the place where I moved it from:
```rust
fn print_factors_str(
    num_str: &str,
    w: &mut io::BufWriter<impl io::Write>,
    print_exponents: bool,
) -> io::Result<()> {
    let x = match num_str.trim().parse::<u64>() {
        Ok(x) => x,
        Err(e) => {
            // We return Ok() instead of Err(), because it's non-fatal and we should try the next
            // number.
            show_warning!("{}: {}", num_str.maybe_quote(), e);
            set_exit_code(1);
            return Ok(());
        }
    };
```

Reply:
I recently learned about "let-else" flow control pattern of rust and used it now:

```rust
    let rx = num_str.trim().parse::<num_bigint::BigUint>();
    let Ok(x) = rx else {
        // return Ok(). it's non-fatal and we should try the next number.
        show_warning!("{}: {}", num_str.maybe_quote(), rx.unwrap_err());
        set_exit_code(1);
        return Ok(());
    };
```